src/targets/java/okhttp.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * @description
+ * HTTP code snippet generator for Java using OkHttp.
+ *
+ * @author
+ * @shashiranjan84
+ *
+ * for any questions or issues regarding the generated code snippet, please open an issue mentioning the author.
+ */
+
+'use strict'
+
+var util = require('util')
+var CodeBuilder = require('../../helpers/code-builder')
+
+module.exports = function (source, options) {
+  var opts = util._extend({
+    indent: '  '
+  }, options)
+
+  var code = new CodeBuilder(opts.indent)
+
+  var methods = [ 'GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'HEAD']
+
+  var methodsWithBody = [ 'POST', 'PUT', 'DELETE', 'PATCH']
```

Comment:
OkHttp does not support get and head with body plus other methods 'POST',
'PUT', 'DELETE', 'PATCH' need to have body or passed null to api.


http://square.github.io/okhttp/javadoc/index.html

On Sun, Mar 29, 2015, 1:01 AM Ahmad Nassri <notifications@github.com> wrote:

> In src/targets/java/okhttp.js
> <https://github.com/Mashape/httpsnippet/pull/60#discussion_r27352431>:
>
> > +
> > +'use strict'
> > +
> > +var util = require('util')
> > +var CodeBuilder = require('../../helpers/code-builder')
> > +
> > +module.exports = function (source, options) {
> > +  var opts = util._extend({
> > +    indent: '  '
> > +  }, options)
> > +
> > +  var code = new CodeBuilder(opts.indent)
> > +
> > +  var methods = [ 'GET', 'POST', 'PUT', 'DELETE', 'PATCH', 'HEAD']
> > +
> > +  var methodsWithBody = [ 'POST', 'PUT', 'DELETE', 'PATCH']
>
> isn't this redundant? all methods could technically have bodies! should
> just focus on whether the postData has content instrlead.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Mashape/httpsnippet/pull/60/files#r27352431>.
>